APIMatic.Core.Test/AuthenticationTest.cs

```diff
@@ -0,0 +1,137 @@
+            var globalConfiguration = new GlobalConfiguration.Builder()
+                .ServerUrls(new Dictionary<Enum, string>
+                {
+                    {MockServer.Server1, "http://my/path:3000/{one}"},
+                }, MockServer.Server1)
+                .AuthManagers(new Dictionary<string, AuthManager>()
+                {
+                    {"basic", basicAuthManager},
+                    {"header", new HeaderAuthManager("my api key", "test token")},
+                    {"query", new QueryAuthManager("my api key", "test token")}
+                })
+                .HttpConfiguration(_clientConfiguration)
+                .Build();
+
+            var request = globalConfiguration.GlobalRequestBuilder()
+                .Setup(HttpMethod.Get, "/auth")
+                .WithOrAuth(auth => auth
+                    .Add("basic")
+                    .AddAndGroup(innerGroup => innerGroup
+                        .Add("header")
+                        .Add("query")))
+                .Build();
+
+            Assert.AreEqual(basicAuthManager.GetBasicAuthHeader(), request.Headers["Authorization"]);
+            Assert.AreEqual("my api key", request.Headers["API-KEY"]);
+            Assert.AreEqual("test token", request.Headers["TOKEN"]);
+            Assert.AreEqual("my api key", request.QueryParameters["API-KEY"]);
+            Assert.AreEqual("test token", request.QueryParameters["TOKEN"]);
+        }
+
+        [Test]
+        public void Multiple_Authentication_OR_Validation_Failure()
+        {
+            var globalConfiguration = new GlobalConfiguration.Builder()
+                .ServerUrls(new Dictionary<Enum, string>
+                {
+                    {MockServer.Server1, "http://my/path:3000/{one}"},
+                }, MockServer.Server1)
+                .AuthManagers(new Dictionary<string, AuthManager>()
+                {
+                    {"basic", new BasicAuthManager("username", null)},
+                    {"header", new HeaderAuthManager("my api key", null)},
+                    {"query", new QueryAuthManager("my api key", "test token")}
+                })
+                .HttpConfiguration(_clientConfiguration)
+                .Build();
+
+            var exp = Assert.Throws<ArgumentNullException>(() => globalConfiguration.GlobalRequestBuilder()
+                .Setup(HttpMethod.Get, "/auth")
+                .WithOrAuth(auth => auth
+                    .Add("basic")
+                    .AddAndGroup(innerGroup => innerGroup
+                        .Add("header")
+                        .Add("query")))
+                .Build());
+
+            Assert.AreEqual("Following authentication credentials are required:\n" +
+                "-> Missing required header field: Authorization\n" +
+                "-> Missing required header field: TOKEN", exp.Message);
+        }
+
+        [Test]
+        public void Multiple_Authentication_AND_Validation_Failure()
```

Comment:
I'd recommend adding another case of and validation failure where all auth managers are invalid so that we could have a test case to confirm that all validation failures are captured.

Reply:
Added the suggested unit test